### PR TITLE
Update sig-clients README

### PR DESCRIPTION
**Pull request by znewman01**
Please take a look! Note that much of this is provisional (e.g., the roadmap, the mission) and so you shouldn't worry about it for now, we'll discuss at the next meeting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,106 @@
...
+- [sigstore-conformance](https://github.com/sigstore/sigstore-conformance): conformance test suites for clients
+- [architecture-docs](https://github.com/sigstore/architecture-docs]): specifications
+- Sigstore client implementations in various languages:
+  - Go: [sigstore-go](https://github.com/sigstore/sigstore-go), [Cosign](https://github.com/sigstore/cosign), [Gitsign](https://github.com/sigstore/gitsign),
+  - Java: [sigstore-java](https://github.com/sigstore/sigstore-java)
+  - JavaScript: [sigstore-js](https://github.com/sigstore/sigstore-js)
+  - Python: [sigstore-python](https://github.com/sigstore/sigstore-python)
+  - Ruby: [sigstore-ruby](https://github.com/sigstore/sigstore-ruby)
+  - Rust: [sigstore-rs](https://github.com/sigstore/sigstore-rs)
+[policy-controller](https://github.com/sigstore/policy-controller)
```
Comment (Hayden-IO) [inline README.md]: Is policy controller in scope? This seems like more than just a sigstore client

File: README.md [hunk 2]
```diff
@@ -1,6 +1,106 @@
...
+
+- sig-clients meeting (monthly; [notes](https://docs.google.com/document/d/1PNbBZSG3QC8hWVYBx6YDppaXwmSLDfx7t66ECaGa8y4/edit#heading=h.hb09y2f8i7to))
+- Sigstore Java (weekly; [notes](https://docs.google.com/document/d/1R7mL-IUrc2Z_LuOIvwDWshVuPQS_2VNE_cIQx4Oy5zw/edit))
+- Sigstore Golang (biweekly; [notes](https://docs.google.com/document/d/1EcJIhqSS9E86cHAQXaXiu2_r1s0kNbHz4uLLwwGo-vw/edit#heading=h.td0phy2bwk06))
+
+**Slack.** We also communicate on [Slack](https://links.sigstore.dev/slack-invite). Channels that might be of interest include: `#clients`, `#java`, `#ruby-gems`, `#sigstore-rust`, `#sigstore-go`, `#cosign`.
```
Comment (Hayden-IO) [inline README.md]: add #gitsign?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,106 @@
...
+
+Last updated: 2023-04-17. See also [project-wide roadmap](https://github.com/sigstore/community/blob/main/ROADMAP.md) (possibly out-of-date).
+
+*This currently represents @znewman01's brain dump; at the next client meeting, we'll work through this.*
+
+**Short-term (months).**
+
+Goal: focusing on Cosign in the near-term as the most widely-used client. If
+other clients like what we're doing there, they can emulate it. Document
+expectations for clients.
```
Comment (Hayden-IO) [inline README.md]: Just wanted to +1 this for immediate focus for Cosign.

I think it'd also be good to include tasks for the other client languages too

File: README.md [hunk 4]
```diff
@@ -1,6 +1,106 @@
...
+  - Fancier policies
+    - Unification between Cosign and policy-controller
+    - e.g., SBOM https://github.com/sigstore/cosign/issues/1954
+- Testing: [conformance test suite](https://github.com/sigstore/sigstore-conformance)
+  - "End-to-end" tests: hook up to [clients](https://github.com/sigstore/sigstore-conformance/issues/61)
+  - ["Fuzz tests"](https://github.com/sigstore/sigstore-conformance/issues/64)
+
+**Long-term (eventually).**
+
+- [Plumbing and porcelain](https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain) model for Cosign.
```
Comment (Hayden-IO) [inline README.md:97]: What is this referring to?

File: README.md (hunk 1 above)
Comment (kommendorkapten) [inline README.md]:
```suggestion
- [policy-controller](https://github.com/sigstore/policy-controller)
```

File: README.md (hunk 3 above)
Comment (kommendorkapten) [reply README.md]: Mention that the goal is **not** to have each language client being complete replacement for the cosing cli too? I.e the language clients should focus on implementing support for signing and verification for ecosystem adoption. Any needed cli functionality to fully test the client is still encouraged.

File: README.md (hunk 1 above)
Comment (znewman01) [reply README.md]: > more than just a sigstore client

So is Cosign :)

I think "resolve discrepancies between policy-controller and Cosign" is *plausibly* in-scope, and relates to a lot of the near-term UX/policy issues. Given that we're not asserting any authority here over the project, just saying that policy-controller might care about some of the decisions we make, I'm comfortable erring on the side of listing more projects.

File: README.md (hunk 4 above)
Comment (znewman01) [reply README.md:97]: This is related to the issue of untangling Cosign, generally. The goal is that `cosign verify` should just be a chain of commands like `cosign fetch-bundle | cosign format-oci-metadata | cosign verify-blob`.

File: README.md (hunk 3 above)
Comment (znewman01) [reply README.md]: > I think it'd also be good to include tasks for the other client languages too

Added a generic "support other languages" item with a TODO to solicit per-language items at the next meeting.

> Mention that the goal is not to have each language client being complete replacement for the cosing cli too? 

Added a clause about "in-scope for each client".

File: README.md (hunk 2 above)
Comment (znewman01) [reply README.md]: Good catch, done.

**Review by kommendorkapten — APPROVED** [review]

**Review by Hayden-IO — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**